controllers/akodeploymentconfig/akodeploymentconfig_controller_avi_phase.go

```diff
@@ -259,6 +267,99 @@ func (r *AKODeploymentConfigReconciler) reconcileCloudUsableNetwork(
 	return res, nil
 }
 
+func (r *AKODeploymentConfigReconciler) reconcileAviInfraSetting(
+	ctx context.Context,
+	log logr.Logger,
+	adc *akoov1alpha1.AKODeploymentConfig,
+) (ctrl.Result, error) {
+	res := ctrl.Result{}
+
+	log.Info("Start reconciling AVIInfraSetting")
+
+	if adc.Spec.ControlPlaneNetwork.Name == "" {
+		log.Info("ControlPlaneNetwork is empty in akoDeploymentConfig, skip creating AVIInfraSetting")
+		return res, nil
+	}
+
+	newAviInfraSetting := r.createAviInfraSetting(adc)
+
+	if err := r.Get(ctx, client.ObjectKey{
+		Name: haprovider.GetAviInfraSettingName(adc),
+	}, newAviInfraSetting); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("AVIInfraSetting doesn't exist, start creating it")
+			return res, r.Create(ctx, newAviInfraSetting)
+		}
+		log.Error(err, "Failed to get AVIInfraSetting, requeue")
+		return res, err
+	}
+	aviInfraSetting := newAviInfraSetting.DeepCopy()
+	return res, r.Update(ctx, aviInfraSetting)
+}
+
+func (r *AKODeploymentConfigReconciler) createAviInfraSetting(adc *akoov1alpha1.AKODeploymentConfig) *akov1alpha1.AviInfraSetting {
+
+	return &akov1alpha1.AviInfraSetting{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
Do you really need this?

Reply:
I'm not sure when we need this. It might be okay if I remove it here.